apps/designer/app/shared/db/build.server.ts

```diff
@@ -0,0 +1,42 @@
+import { prisma, Build as DbBuild } from "@webstudio-is/prisma-client";
+import { z } from "zod";
+
+const Page = z.object({
+  id: z.string(),
+  name: z.string(),
+  path: z.string(),
+  title: z.string(),
+  meta: z.record(z.string(), z.string()),
+  treeId: z.string(),
+});
+
+type Page = z.infer<typeof Page>;
+
+const Pages: z.ZodType<{ homePage: Page; pages: Array<Page> }> = z.object({
+  homePage: Page,
```

Comment:
why not make it an id? 
```suggestion
  home: Page['id'],
```

Reply:
ah pages won't have the home page? I thought pages will have all the pages, and one of them is home

Reply:
I'm a bit obsessed with the idea of creating data-structures that make invalid states impossible to represent https://www.youtube.com/watch?v=IcgmSRJHu_8

Here, if we go with `id`, it may happen that the pages array will not have a page with that `id`.

But I think maybe we should just make it a simple array after all, with a `boolean` property `Page.isHome`. Need to see how a separate `homePage` complicates or doesn't complicate the code.

Reply:
Will probably go with a simple array + checks for all requirements for home page in the zod schema.

Reply:
> I'm a bit obsessed with the idea of creating data-structures that make invalid states impossible to represent

me too, definitely

> Here, if we go with id, it may happen that the pages array will not have a page with that id.

could be ensured by the logic, but yeah its less robust

Reply:
if you go with an isHome you have to ensure only one has it, if you go with a home: id - you only need to ensure the id still exists in the tree, so user can't delete home page without setting to home some other page

Reply:
Keeping `homePage: Page` for now, may reconsider later